src/analysis/timeline/timelineCredView.test.js

```diff
@@ -0,0 +1,109 @@
+// @flow
+
+import {sum} from "d3-array";
+import sortBy from "lodash.sortby";
+import {utcWeek} from "d3-time";
+import {NodeAddress, Graph} from "../../core/graph";
+import {TimelineCred, type TimelineCredConfig} from "./timelineCred";
+import {TimelineCredView} from "./timelineCredView";
+import {type FilteredTimelineCred} from "./filterTimelineCred";
+import {defaultWeights} from "../weights";
+
+describe("src/analysis/timeline/timelineCred", () => {
+  const credConfig: () => TimelineCredConfig = () => ({
+    scoreNodePrefix: NodeAddress.fromParts(["foo"]),
+    filterNodePrefixes: [NodeAddress.fromParts(["foo"])],
+    types: {nodeTypes: [], edgeTypes: []},
+  });
+
+  function exampleTimelineCred(): TimelineCred {
+    const startTimeMs = +new Date(2017, 0);
+    const endTimeMs = +new Date(2017, 6);
+    const boundaries = utcWeek.range(startTimeMs, endTimeMs);
+    const intervals = [];
+    for (let i = 0; i < boundaries.length - 1; i++) {
+      intervals.push({
+        startTimeMs: +boundaries[i],
+        endTimeMs: +boundaries[i + 1],
+      });
+    }
+    const users = [
+      ["starter", (x) => Math.max(0, 20 - x)],
+      ["steady", (_) => 4],
+      ["finisher", (x) => (x * x) / 20],
+      ["latecomer", (x) => Math.max(0, x - 20)],
+    ];
+
+    const graph = new Graph();
+    const addressToCred = new Map();
+    for (const [name, generator] of users) {
+      const address = NodeAddress.fromParts(["foo", name]);
+      graph.addNode({
+        address,
+        description: `[@${name}](https://github.com/${name})`,
+        timestampMs: null,
+      });
+      const scores = intervals.map((_unuesd, i) => generator(i));
+      addressToCred.set(address, scores);
+    }
+    const filteredTimelineCred: FilteredTimelineCred = {
+      intervals,
+      addressToCred,
+    };
+    const params = {alpha: 0.05, intervalDecay: 0.5, weights: defaultWeights()};
+    return new TimelineCred(graph, filteredTimelineCred, params, credConfig());
+  }
+
+  it("JSON serialization works", () => {
+    const json = exampleTimelineCred().toJSON();
+    const tc_ = TimelineCred.fromJSON(json, credConfig());
+    const tcv = new TimelineCredView(exampleTimelineCred());
+    const tcv_ = new TimelineCredView(tc_);
+    expect(tcv.credSortedNodes(NodeAddress.empty)).toEqual(
```

Comment:
the JSON serialization of TimelineCred is already tested in timelineCred.js. And timelineCredView doesn't implement any new JSON serialization. so, we don't need this test.